src/toil/batchSystems/slurm.py

```diff
@@ -68,9 +69,61 @@
     "STAGE_OUT",
     "STOPPED",
     "SUSPENDED"
-} 
+}
 
 class SlurmBatchSystem(AbstractGridEngineBatchSystem):
+    class PartitionInfo(NamedTuple):
+        partition_name: str
+        gres: bool
+        time: str
+        priority: str
+        cpus: str
+        memory: str
+
+    class PartitionSet:
+        """
+        Set of available partitions detected on the slurm batch system
+        """
+        default_gpu_partition: Optional[SlurmBatchSystem.PartitionInfo]
+        all_partitions: List[SlurmBatchSystem.PartitionInfo]
+        gpu_partitions: List[SlurmBatchSystem.PartitionInfo]
```

Comment:
This should probably be a set of names, since that's how we use it later.
```suggestion
        gpu_partitions: Set[str]
```